### PR TITLE
PCLOCK / MCLOCK - implementations of POSIX and monotonic clocks

### PR DESCRIPTION
**Pull request by mattgray**
See https://github.com/mirage/mirage/issues/442

### CODE REVIEW

**Comment by mattgray** [comment]
Should I rebase and squash some of these commits? There's a lot of WIP commits here

File: unix/mclock.mli [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+val elapsed_ns : unit -> int64
+val period_ns : unit -> int64 option
```
Comment (hannesm) [inline unix/mclock.mli]: this (and other interfaces) should specifically refer to V1.MCLOCK / V1.PCLOCK (something like `include V1.MCLOCK`)

**Comment by yomimono** [comment]
Sorry for the wait here -- a squash would be nice, but I might take a few liberties and do it myself if you don't have the time in the next couple of days (as long as you don't mind, of course).

**Comment by mattgray** [comment]
I have a few things to add, namely DEVICE implementations for PCLOCK and
MCLOCK - those are done and I'm working thorough mirage-skeleton getting it
to compile. So if you want to hold off I can complete those and and then
squash? I hope to get skeleton building by the weekend, but at very least I
can push my changes here tomorrow

**Comment by yomimono** [comment]
Ah, I thought the `DEVICE` implementations were already here in https://github.com/mirage/mirage-clock/pull/7/commits/ab8dd35c8ff1ad2fc67c99cb00cd86be6b3f33ea and https://github.com/mirage/mirage-clock/pull/7/commits/918faa67f5fab02c037355e90896a1821d3edddb ?

**Comment by mattgray** [comment]
Ah did I push them already? Not at a computer so u can't check but if they
look legit then I probably did :)

**Comment by mattgray** [comment]
s/u can't check/I can't check

**Comment by yomimono** [comment]
They look legit to me; I'm working through downstream changes in mirage-logs and we'll see what `make test` says :P

**Pull request merged**